### PR TITLE
feat: add optional headers param to network image

### PR DESCRIPTION
**Pull request by Alexays**
Hey, here's a feature that was missing so I add it.
Open to review :)

### CODE REVIEW

File: packages/flutter/lib/src/services/image_provider.dart [hunk 1]
```diff
@@ -411,6 +411,9 @@ class NetworkImage extends ImageProvider<NetworkImage> {
   /// The scale to place in the [ImageInfo] object of the image.
   final double scale;
 
+  /// The headers that will be used to fech image from network
```
Comment (Hixie) [inline packages/flutter/lib/src/services/image_provider.dart]: fech -> fetch
period at end of sentence

Comment (Hixie) [reply packages/flutter/lib/src/services/image_provider.dart]: Maybe this should be clarified to say HTTP headers, and to point to [HttpClient.get] explicitly and say that it's passed through to that method unmodified?

File: packages/flutter/lib/src/widgets/image.dart [hunk 2]
```diff
@@ -142,7 +142,8 @@ class Image extends StatefulWidget {
     this.matchTextDirection: false,
     this.gaplessPlayback: false,
     this.package,
-  }) : image = new NetworkImage(src, scale: scale),
+    Map<String, String> headers,
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/image.dart]: the docs for this constructor should mention the headers argument too.

**Comment by Alexays** [comment]
Done @Hixie ;)

**Comment by Hixie** [comment]
This is great. Can you add a test? You should be able to create a mock HttpClient object by providing a createHttpClient implementation -- if you look through the exists tests you should be able to find one that does this already. Then you can just mock the "get" method and check that the headers make it there.

File: packages/flutter/lib/src/widgets/image.dart (hunk 2 above)
Comment (Hixie) [inline packages/flutter/lib/src/widgets/image.dart]: the docs for this constructor should mention the headers argument too.

File: packages/flutter/lib/src/services/image_provider.dart [hunk 3]
```diff
@@ -411,6 +411,9 @@ class NetworkImage extends ImageProvider<NetworkImage> {
   /// The scale to place in the [ImageInfo] object of the image.
   final double scale;
 
+  /// The HTTP headers that will be used with [HttpClient.get] to fetch image from network
```
Comment (Hixie) [inline packages/flutter/lib/src/services/image_provider.dart]: still missing the final period. :-)

File: packages/flutter/lib/src/widgets/image.dart [hunk 4]
```diff
@@ -128,6 +128,7 @@ class Image extends StatefulWidget {
   /// Creates a widget that displays an [ImageStream] obtained from the network.
   ///
   /// The [src], [scale], and [repeat] arguments must not be null.
+  /// An optional [headers] argument can be used to use custom HTTP headers
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/image.dart]: Also here.

File: packages/flutter/lib/src/widgets/image.dart (hunk 2 above)
Comment (Alexays) [reply packages/flutter/lib/src/widgets/image.dart]: I did it or I don't understand, I added a test however @Hixie

**Comment by Hixie** [comment]
Thanks for your contribution!

**Pull request merged**